api/v1beta2/awsmachine_types.go

```diff
@@ -188,7 +200,11 @@ type AWSMachineSpec struct {
 	// Tenancy indicates if instance should run on shared or single-tenant hardware.
 	// +optional
 	// +kubebuilder:validation:Enum:=default;dedicated;host
-	Tenancy string `json:"tenancy,omitempty"`
+	Tenancy Tenancy `json:"tenancy,omitempty"`
```

Comment:
By reading this API is not clear to me which input should I set for tenancy host with autoplacement, can we document that here?

Reply:
thanks! "If host tenancy is used and auto-placement is disabled for the Dedicated Host, It is required to set '.spec.hostPlacment.hostID'" though the requirement is actually not enforced. Can we document what happens if not set in that case, e.g. it's required or no host would be picked. Can we also document all matrix combinations this API enables: If both hostID and autoplacement are set, the hostID would take precedence? If the hostID doesn't exist would it fallback to autoplacement?

Reply:
we can't enforce it, because we don't know if auto-placement is enabled or not on the dedicated Host, since its created directly by the user out of band.

Reply:
updated.